xmppserver/src/main/java/org/jivesoftware/openfire/group/DefaultGroupPropertyMap.java

```diff
@@ -389,15 +414,16 @@ public boolean hasNext() {
         public void remove() {
             delegate.remove();
             K key = current.getKey();
+            V value = current.getValue();
             if (key instanceof String) {
-                deleteProperty((String)key);
+                deleteProperty((String)key, value instanceof String ? (String)value : null);
```

Comment:
(there's clearly a lot of String protection going on here that I don't understand)

Reply:
This implementation apparently only works with Strings, while it accepts a generic type. Something to dig into another time...